net/graphsync_fetcher.go

```diff
@@ -185,3 +237,60 @@ func (gsf *GraphSyncFetcher) loadTipsetFromCids(ctx context.Context, cids []cid.
 
 	return types.NewTipSet(validatedBlocks...)
 }
+
+func (gsf *GraphSyncFetcher) missingCids(cids []cid.Cid) ([]cid.Cid, error) {
+	missing := make([]cid.Cid, 0, len(cids))
+	for _, cid := range cids {
+		hasBlock, err := gsf.store.Has(cid)
+		if err != nil {
+			return nil, err
+		}
+		if !hasBlock {
+			missing = append(missing, cid)
+		}
+	}
+	return missing, nil
+}
+
+func (gsf *GraphSyncFetcher) areCidsMissing(tsKey types.TipSetKey, rpf *requestPeerFinder) (bool, error) {
+	missing, err := gsf.missingCids(tsKey.ToSlice())
+	if err != nil {
+		return false, err
+	}
+	if len(missing) != 0 {
+		err := rpf.FindNextPeer()
```

Comment:
Blocking: I have a strong preference to put this on line 158.  This is a hidden side effect I didn't expect from the name.